File: app/src/main/java/de/garten/training/depotflow/domain/SyncStatus.java
package de.garten.training.depotflow.domain;

public enum SyncStatus {
    CLEAN("clean"),
    DIRTY("dirty"),
    SYNCING("syncing"),
    FAILED("failed");

    private final String databaseValue;

    SyncStatus(String databaseValue) {
        this.databaseValue = databaseValue;
    }

    public String toDatabaseValue() {
        return databaseValue;
    }

    public static SyncStatus fromDatabaseValue(String value) {
        if (value == null) {
            return CLEAN;
        }
        for (SyncStatus status : values()) {
            if (status.databaseValue.equals(value)) {
                return status;
            }
        }
        return FAILED;
    }
}

File: app/src/main/java/de/garten/training/depotflow/domain/SyncStatus.kt
package de.garten.training.depotflow.domain

enum class SyncStatus(private val databaseValue: String) {
    CLEAN("clean"),
    DIRTY("dirty"),
    SYNCING("syncing"),
    FAILED("failed");

    fun toDatabaseValue(): String = databaseValue

    companion object {
        @JvmStatic
        fun fromDatabaseValue(value: String?): SyncStatus {
            if (value == null) {
                return CLEAN
            }
            return entries.firstOrNull { it.databaseValue == value } ?: FAILED
        }
    }
}
